### PR TITLE
*: split statistics from core package

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Part of modularizing PD. 

### What is changed and how it works?
This PR splits statistics from `core` package.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/api/trend.go [hunk 1]
```diff
@@ -18,9 +18,10 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/pingcap/pd/server/statistics"
+
```
Comment (disksing) [inline server/api/trend.go]: Remove empty line.

File: server/cluster.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/pingcap/pd/server/statistics"
+
```
Comment (disksing) [inline server/cluster.go]: remove empty line.

File: server/cluster.go [hunk 3]
```diff
@@ -326,12 +328,19 @@ func (c *RaftCluster) GetStoreRegions(storeID uint64) []*core.RegionInfo {
...
-func (c *RaftCluster) GetRegionStats(startKey, endKey []byte) *core.RegionStats {
+func (c *RaftCluster) GetRegionStats(startKey, endKey []byte) *statistics.RegionStats {
 	c.RLock()
 	defer c.RUnlock()
 	return c.cachedCluster.getRegionStats(startKey, endKey)
 }
 
+// GetStoresStats returns stores' statistics from cluster.
+func (c *RaftCluster) GetStoresStats() *statistics.StoresStats {
+	c.Lock()
```
Comment (disksing) [inline server/cluster.go]: Can we use `Rlock`?

File: server/coordinator.go [hunk 4]
```diff
@@ -19,6 +19,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/pingcap/pd/server/statistics"
+
```
Comment (disksing) [inline server/coordinator.go]: remove empty line

File: server/store_statistics.go [hunk 5]
```diff
@@ -17,6 +17,8 @@ import (
 	"fmt"
 	"strconv"
 
+	"github.com/pingcap/pd/server/statistics"
+
```
Comment (disksing) [inline server/store_statistics.go]: remove empty line.

File: server/statistics/region.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
+package statistics
```
Comment (nolouch) [inline server/statistics/region.go]: license.

**Pull request merged**